exp/query/api.go

```diff
@@ -79,21 +108,154 @@ func (a *API) Search(ctx context.Context, req *SearchRequest) (*SearchResponse,
 		return nil, err
 	}
 
-	// Expectations
-	// 0. Check if tenant status == FROZEN
-	// 1. Build local path `/<datapath>/<collection>/<tenant>` from search request
-	// 2. If it doesn't exist, fetch from object store to same path as above
-	// 3. If it doesn't exist on object store, return 404
-	// 4. Once in the local disk, load the index
-	// 5. Searve the search.
-	return &SearchResponse{}, nil
+	localPath := path.Join(a.offload.DataPath, strings.ToLower(req.Collection), strings.ToLower(req.Tenant), defaultLSMRoot)
+
+	// TODO(kavi): Avoid creating store every time?
```

Comment:
Maybe store the store in an async map and have a compare and swap on it for concurrent requests ? Would allow us to re-use it or create it if not present (on startup for example).

Reply:
Turns out we need this store to be created everytime because local path changes with incoming `tenant` and `collection` strings.

One thing I added to do selective download. So avoid downloading from s3 all the time if the correct path exists. And Nate's data version will make sure that path is up to date.